core-bundle/src/Routing/AbstractPageRouteProvider.php

```diff
@@ -167,8 +167,8 @@ protected function compareRoutes(Route $a, Route $b, array $languages = null): i
             return 1;
         }
 
-        $pathA = $a instanceof PageRoute ? substr($a->getPath(), 0, -\strlen($a->getUrlSuffix())) : $a->getPath();
-        $pathB = $b instanceof PageRoute ? substr($b->getPath(), 0, -\strlen($b->getUrlSuffix())) : $b->getPath();
+        $pathA = $a instanceof PageRoute && $a->getUrlSuffix() ? substr($a->getPath(), 0, -\strlen($a->getUrlSuffix())) : $a->getPath();
+        $pathB = $b instanceof PageRoute && $b->getUrlSuffix() ? substr($b->getPath(), 0, -\strlen($b->getUrlSuffix())) : $b->getPath();
```

Comment:
This could also work without calling `getUrlSuffix()` twice:

```suggestion
        $pathA = $a instanceof PageRoute ? substr($a->getPath(), 0, -\strlen($a->getUrlSuffix()) ?: null) : $a->getPath();
        $pathB = $b instanceof PageRoute ? substr($b->getPath(), 0, -\strlen($b->getUrlSuffix()) ?: null) : $b->getPath();
```

Reply:
technically true, but we never want two inline comparison on the same line

Reply:
What exactly do you mean?

Reply:
There would be two `… ? … : …` on the same line, which is confusing and reported by EA-inspections (and probably other tools)

Reply:
Why would it be fine to use multiple null coalescing operators but not elvis? I think the idea of these rules is that you should avoid nesting ternary operators where it would be hard to spot their associativity.